packages/mobile/src/qrcode/utils.ts

```diff
@@ -64,6 +69,35 @@ export function* handleBarcode(
     yield put(showError(ErrorMessages.QR_FAILED_INVALID_ADDRESS))
     return
   }
+  try {
+    if (secureSendTxData) {
+      if (!secureSendTxData.recipient.e164PhoneNumber) {
+        throw Error(`Invalid recipient type for Secure Send: ${secureSendTxData.recipient.kind}`)
+      }
+
+      const userScannedAddress = data.address.toLowerCase()
+      const { e164PhoneNumber } = secureSendTxData.recipient
+      const possibleRecievingAddresses = e164NumberToAddress[e164PhoneNumber]
+      // This should never happen. Secure Send is triggered when there are
+      // mutliple addrresses for a given phone number
+      if (!possibleRecievingAddresses) {
+        throw Error("No addresses associated with recipient's phone number")
+      }
+
+      const possibleRecievingAddressesFormatted = possibleRecievingAddresses.map((address) =>
+        address.toLowerCase()
+      )
+      if (!possibleRecievingAddressesFormatted.includes(userScannedAddress)) {
+        yield put(showError(ErrorMessages.QR_FAILED_INVALID_RECIPIENT))
+        return
+      }
+
+      yield put(validateRecipientAddressSuccess(e164PhoneNumber, userScannedAddress))
+    }
+  } catch (error) {
+    Logger.error(TAG + '@handleBarcode', `Error with Secure Send: `, error)
```

Comment:
Should this saga proceed if there's an error?

Reply:
Nope, this should return. Good catch